### PR TITLE
Set default properties for OpenSSL command-line programs

### PR DESCRIPTION
**Pull request by olszomal**
OpenSSL command-line programs should set default properties for all future EVP algorithm fetches.
The TLS_CHACHA20_POLY1305_SHA256 ciphersiute is disabled in the FIPS mode, but openssl-ciphers supports this ciphersiute in the FIPS mode:
```
$ openssl ciphers -tls1_3 -s -v -provider fips -propquery fips=yes
TLS_AES_256_GCM_SHA384         TLSv1.3 Kx=any      Au=any   Enc=AESGCM(256)            Mac=AEAD
TLS_CHACHA20_POLY1305_SHA256   TLSv1.3 Kx=any      Au=any   Enc=CHACHA20/POLY1305(256) Mac=AEAD
TLS_AES_128_GCM_SHA256         TLSv1.3 Kx=any      Au=any   Enc=AESGCM(128)            Mac=AEAD
```
Expected result is:
```
$ openssl ciphers -tls1_3 -s -v -provider fips -propquery fips=yes
TLS_AES_256_GCM_SHA384         TLSv1.3 Kx=any      Au=any   Enc=AESGCM(256)            Mac=AEAD
TLS_AES_128_GCM_SHA256         TLSv1.3 Kx=any      Au=any   Enc=AESGCM(128)            Mac=AEAD
```

CLA: regular

### CODE REVIEW

File: apps/lib/app_libctx.c [hunk 1]
```diff
@@ -12,9 +12,10 @@
 static OSSL_LIB_CTX *app_libctx = NULL;
 static const char *app_propq = NULL;
 
-int app_set_propq(const char *arg)
+int app_set_propq(OSSL_LIB_CTX *libctx, const char *arg)
 {
     app_propq = arg;
+    EVP_set_default_properties(libctx, app_propq);
```
Comment (t8m) [inline apps/lib/app_libctx.c:18]: Should the return value be checked here?

**Review by t8m — CHANGES_REQUESTED** [review]

**Comment by paulidale** [comment]
I think it would make more sense to remove `app_propq` and `app_get0_propq()` completely and rely on just the default query being set.  Passing the same thing for the query as what the default query is set to, is harmless but wasteful.

**Comment by t8m** [comment]
I think what @paulidale said above makes sense.

**Review by paulidale — APPROVED** [review]

**Comment by mattcaswell** [comment]
There is a subtle change of behaviour here.

Prior to this PR we pass the prop query directly to all fetches (as is done extensively in the apps, i.e. most places where we call `app_get0_propq`). If there is an existing default prop query picked up from a config file, then it is merged with the fetch specific propq (as supplied to a `-propquery` command line argument) and *both* apply.

After this PR if there is a default prop query picked up from the config file then it is completely overridden by the command line `-propquery` argument. *Only* the command line supplied propquery applies.

This could be breaking.

**Comment by DDvO** [comment]
@olszomal, I wonder why you closed your PR without comment?
Was this by mistake?

**Comment by olszomal** [comment]
Fixed by #17595

**Pull request closed without merging**